### ISSUE TITLE
EDDI crashes when clicking on "get route" or "next in route"

### ISSUE THREAD

**Issue by netchild** [EDCD/EDDI#2341]
Hi,

take 2 passenger mission with more than 1 destination each, click on "get route" or "next in route". EDDI crashes.

Bye,
Alexander.

**Comment by Tkael** [comment]
Please provide your log files from `%APPDATA%/EDDI` to help identify the source of the crash?

Please also review the bug report issue template and add missing information from that template to this ticket? See https://github.com/EDCD/EDDI/blob/develop/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by netchild** [comment]
[eddi1.log](https://github.com/EDCD/EDDI/files/8671086/eddi1.log)
[eddi3.log](https://github.com/EDCD/EDDI/files/8671105/eddi3.log)

Version 4.0.1
Language set to English.

**Comment by Tkael** [comment]
EDDI appears to be having a problem setting the contents of your clipboard. Since the clipboard method is tied into the front-end UI in version 4.0.1, an exception there seems to be cascading and causing the UI thread and ultimately the entire application to fail.

It's not clear to me why EDDI is failing to set the clipboard. What Windows version are you running? Are you aware of anything in your Windows configuration that might prevent an application from setting the clipboard?

This issue should be corrected by the next release, which separates the clipboard method from the front-end UI and puts it more directly under your control. In the meantime, if we're unable to determine why EDDI is unable to set the clipboard then I'd suggest that you do not use the `get route` or `next in route` UI buttons.

**Comment by netchild** [comment]
Windows 10 with a microsoft account. Clipboard settings are NOT set to sync into the cloud / sync to all devices. And no clipboard history activated.
No idea what may prevent the setting of the clipboard. Firefox is able to set the clipboard (e.g. the button on Inara to copy the system name to the clipboard).

**Comment by Tkael** [comment]
Intriguing. Similar to my own setup (and likely also similar to that of many others). I'm curious - is the issue repeating every time that you try it or was occurring during one particular game session only? Have you rebooted your PC since then and if so does the issue still occur?

**Comment by netchild** [comment]
That was the first time I tried it. Same ED session, tried several times with restarted EDDI.

Now I tried it again (several reboots since then), and now it doesn't crash but tells it is unable to calculate the missions route... so not really reaching the failure case with the clipboard.
![eddi_missions](https://user-images.githubusercontent.com/3617760/169260796-24ae8813-d591-48a9-9bf3-5f238e2a935d.PNG)
